### PR TITLE
Create filename character restriction

### PR DESCRIPTION
**Pull request by spectranaut**
Only allow alphanumeric characters and `-`, `_`, `.` in the file names of test262 tests. 

This patch updates:
- two files with `$` in name
- CONTRIBUTING.md
- the linter!

This might be too constricting? It was motivate by an unwanted string interpolation in a script in bash (there was a `$1` in a file name, interpolated into an empty string).

But as there are no other files with special characters, and I wasn't sure which characters to allow other than these, I left it at this.

### CODE REVIEW

File: tools/lint/lint.py [hunk 1]
```diff
@@ -50,7 +51,8 @@
     CheckFeatures('features.txt'),
     CheckHarnessFeatures(),
     CheckLicense(),
-    CheckNegative()
+    CheckNegative(),
+    CheckFileName()
```
Comment (jugglinmike) [inline tools/lint/lint.py]: This list was meant to be alphabetical. Mind moving this entry (and fixing the location of `CheckFeatures` while you're at it)?

File: tools/lint/lib/checks/filename.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+_DISALLOWED_PATTERN = re.compile('[^a-zA-Z0-9/\\-_.]')
+
+class CheckFileName(Check):
+    '''Ensure tests have a valid name.'''
+    ID = 'FILENAME'
+
+    def run(self, name, meta, source):
+        if _DISALLOWED_PATTERN.search(name):
+            return "Contains non-alphanumeric or `-`, `_`, '.' characters."
+        else:
```
[the remark this answers is not in the document]
Comment (spectranaut) [reply tools/lint/lib/checks/filename.py]: haha, so I can! :)

**Comment by spectranaut** [comment]
Whoops about delete file -- but yeah it's pretty cute that the linter caught that! :) Fixed all your requests.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 Test cases should be created in files that are named to identify the feature or API that's being tested.
 
-There is no strict naming convention. The file names should be human readable, helpful and, ideally, consistent within a single directory. For examples:
+The names should use alphanumeric characters and `.`, `-`, `_`. Otherwise, there is no strict naming convention, but the file names should be human readable, helpful and, ideally, consistent within a single directory. For examples:
```
Comment (rwaldron) [inline CONTRIBUTING.md]: This will be nice to get cleaned up

**Comment by jugglinmike** [comment]
This is almost there! It's still missing a test, though. Let me know if you have any trouble finding how to do that.

**Comment by rwaldron** [comment]
@spectranaut any update on @jugglinmike's request?

**Comment by spectranaut** [comment]
Oh thanks for the reminder @rwaldron ! It dropped of my radar. Will do today :)

**Comment by spectranaut** [comment]
Test submitted, @jugglinmike ! :)

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit 40883f4c6ae17e073dfd03e34f73a6bf8c855595

**Pull request closed without merging**